src/transaction.rs

```diff
@@ -231,11 +237,17 @@ impl Transaction {
 
     #[cfg(feature = "std")]
     pub fn check_predicate_owner(&self, idx: usize) -> bool {
```

Comment:
Hmm, it seems we don't have any tests for predicate owner validation here or in the VM :/

Reply:
Since we are fixing behavior that wasn't properly unit tested before, I think we should include unit tests to cover the predicate ownership checks with this fix.